src/pages/VenuePage/VenuePage.tsx

```diff
@@ -195,25 +143,22 @@ const VenuePage: React.FC = () => {
     )
       return;
     if (!venue || !user) return;
-    updateProfileEnteredVenueIds(
-      profile?.enteredVenueIds,
-      user?.uid,
-      venue?.id
-    );
+    updateProfileEnteredVenueIds(profile?.enteredVenueIds, userId, venue?.id);
```

Comment:
`updateProfileEnteredVenueIds` is defined in  `src/utils/profile.ts`

This function could be refactored to not require `enteredVenueIds` to be passed into it, and to instead use our 'best practice' firestore update patterns:

- https://firebase.google.com/docs/firestore/manage-data/add-data#update_fields_in_nested_objects
- https://firebase.google.com/docs/firestore/manage-data/add-data#update_elements_in_an_array
  - `admin.firestore.FieldValue.arrayUnion`

From there, this `useEffect`'s deps can be simplified down to only being the `userId` and `venueId` (rather than the whole objects), which will cause the code to only ever run when the exact things we care about have changed.

This will simplify the code, make it more robust, and make it more performant.